File: src/main/java/com/procurement/notice/dto/Criterion.java
package com.procurement.notice.dto;

import java.util.List;
import lombok.Data;

@Data
public class Criterion {
    public String id;
    public String title;
    public String description;
    public String source;
    public String relatesTo;
    public String relatedItem;
    public List<RequirementGroup> requirementGroups;
}

File: src/main/kotlin/com/procurement/notice/model/ocds/Criterion.kt
package com.procurement.notice.model.ocds

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonInclude

data class Criterion @JsonCreator constructor(

    @field:JsonInclude(JsonInclude.Include.NON_NULL)
    val id: String?,

    @field:JsonInclude(JsonInclude.Include.NON_NULL)
    val title: String?,

    @field:JsonInclude(JsonInclude.Include.NON_NULL)
    val description: String?,

    @field:JsonInclude(JsonInclude.Include.NON_NULL)
    val source: String?,

    @field:JsonInclude(JsonInclude.Include.NON_NULL)
    val relatesTo: String?,

    @field:JsonInclude(JsonInclude.Include.NON_NULL)
    val relatedItem: String?,

    @field:JsonInclude(JsonInclude.Include.NON_EMPTY)
    val requirementGroups: HashSet<RequirementGroup>?
)
